azure-client-runtime/src/main/java/com/microsoft/azure/AzureClient.java

```diff
@@ -154,7 +154,8 @@ public void setLongRunningOperationRetryTimeout(int longRunningOperationRetryTim
      * @return the observable of which a subscription will lead PUT or PATCH action.
      */
     public <T> Single<PollingState<T>> beginPutOrPatchAsync(Observable<Response<ResponseBody>> observable, final Type resourceType) {
-        return observable.map(new Func1<Response<ResponseBody>, PollingState<T>>() {
+        return observable
+                .observeOn(Schedulers.immediate()).map(new Func1<Response<ResponseBody>, PollingState<T>>() {
```

Comment:
What does scheduling on the immediate scheduler do differently than the default?